### ISSUE TITLE
TypeScript Support

### ISSUE THREAD

**Issue by tonyxiao** [react-native-community/react-native-hooks#22]
There's currently no index.d.ts file or corresponding `@types/react-native-hooks` available.

**Comment by tonyxiao** [comment]
What I currently have in my repo in case it's useful as a starting place
```ts
// eslint-disable-next-line eslint-comments/disable-enable-pair
/* eslint-disable @typescript-eslint/no-explicit-any */
declare module 'react-native-hooks' {
  import {AppState, AccessibilityInfo, NetInfo} from 'react-native'

  export function useAccessibilityInfo(): AccessibilityInfo

  export function useAppState(): AppState

  /**
   * const [photos, getPhotos, saveToCameraRoll] = useCameraRoll()
   * {
   *   photos.map((photo, index) => /* render photos *\/)
   * }
   * <Button title='Get Photos' onPress={() => getPhotos()}>Get Photos</Button>
   */
  export function useCameraRoll(): [any[], Function, Function]

  export function useClipboard(): [string, (data: string) => void]

  export function useDimensions(): any

  /** const [position, stopObserving, setRNConfiguration] = useGeolocation() */
  export function useGeolocation(): [any, Function, Function]

  export function useNetInfo(): NetInfo

  /**
   * const keyboard = useKeyboard()
   *
   * console.log('keyboard show: ', keyboard.show)
   * console.log('keyboard height: ', keyboard.height)
   */
  export function useKeyboard(): any

  /**
   * const interactionReady = useInteractionManager()
   * console.log('interaction ready: ', interactionReady)
   */
  export function useInteractionManager(): any

  /**
   * const orientation = useDeviceOrientation()
   * console.log('is orientation portrait: ', orientation.portrait)
   * console.log('is orientation landscape: ', orientation.landscape)
   */
  export function useDeviceOrientation(): any
}

```

**Comment by DylanVann** [comment]
This library would be super easy to convert to TypeScript.

**Comment by victorlevasseur** [comment]
We could provide typescript typings or convert the library to Typescript. I can help with that and even help doing the conversion.

**Comment by paldepind** [comment]
#35 closes this. The type definitions are great.

However, my five cents is that it's probably easier to port the library to TS than to maintain separate type definitions.

**Comment by slorber** [comment]
I'd rather have all this reimplemented to TS and having a transpilation step (which could be useful for hooks that work in RNW for example) (bob is easy to use: https://github.com/react-native-community/bob)

**Comment by pvinis** [comment]
We have this one #41, any comments/help is appreciated.

**Comment by ScreamZ** [comment]
Just add `"declaration": true` in `tsco,fig.json` should resolve most things isn't ?

**Comment by luisnaranjo733** [comment]
Looks like #41 was merged on November 22, 2019 but there hasn't been a release of react-navigation-hooks [since August, 2019.](https://www.npmjs.com/package/react-native-hooks)

What needs to be done before we can get another release?

**Comment by luisnaranjo733** [comment]
I just pulled down the unpublished master branch with #41 and as far as I can tell consumers of this project still can't get type safety from this package even with those changes. I can see that the source is now in typescript and it transpiles correctly to JS, but consumers aren't able to import this package succesfully.

> error TS7016: Could not find a declaration file for module 'react-native-hooks'. '/Users/luisnaranjo/code/react-native-hooks/index.js' implicitly has an 'any' type.
  Try `npm install @types/react-native-hooks` if it exists or add a new declaration (.d.ts) file containing `declare module 'react-native-hooks';`

These are the steps I took
1. Clone this project
2. Run yarn install
3. Run yarn build
4. From my project, run npm install /path/to/cloned/source
5. From a TS file in my project: ```import { useDeviceOrientation } from 'react-native-hooks'```

**Comment by pvinis** [comment]
Hm. Could you try adding this:

> Just add `"declaration": true` in `tsconfig.json` should resolve most things isn't ?

and report back? If that's all, let's add that and do a release.

**Comment by Naturalclar** [comment]
@pvinis  `declatation:true` was already in the tsconfig, so I'm not sure if that was the problem.
It looks like `index.js` was not part of the src, so it's type definition `index.d.ts` was not created when running `tsc`.

I've made a PR #52 that will solve the issue.

**Comment by luisnaranjo733** [comment]
Just pulled down @Naturalclar's branch and can confirm that it fixes the problem.

**Comment by pvinis** [comment]
Opssssss 😅 I didn't rename the index file 🤦‍♂. Thanks to both of you for fixing and testing.

**Comment by pvinis** [comment]
Shall we make a new release and close this then?

**Comment by safaiyeh** [comment]
@pvinis @Naturalclar any update on the release of this?

**Comment by pvinis** [comment]
Working on it.

**Comment by LinusU** [comment]
Version 2.0.0 has been released as `@react-native-community/hooks` and I have just verified that the typings are correctly shipped with the package 🚀

**Comment by pvinis** [comment]
I forgot to write here after I released 🤦‍♂. Thanks to everyone involved, from pushing us to do it, all the way to the release!
